sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -33,7 +33,7 @@ object RDDConversions {
     data.mapPartitions { iterator =>
       val numColumns = outputTypes.length
       val mutableRow = new GenericInternalRow(numColumns)
-      val converters = outputTypes.map(CatalystTypeConverters.createToCatalystConverter)
+      val converters = outputTypes.map(CatalystTypeConverters.createToCatalystConverter).toArray
```

Comment:
shall we use `ExpressionEncoder` here?

Reply:
It is a good suggestion, and has been modified, would you like to review it again, thanks.